ktorm-core/src/main/kotlin/org/ktorm/dsl/Query.kt

```diff
@@ -330,24 +329,53 @@ public fun ColumnDeclaring<*>.desc(): OrderByExpression {
     return OrderByExpression(asExpression(), OrderType.DESCENDING)
 }
 
+/**
+ * Specify the pagination limit parameter of this query.
+ *
+ * This function requires a dialect enabled, different SQLs will be generated with different dialects.
+ *
+ * Note that if [limit] is zero then it will be ignored.
+ */
+public fun Query.limit(limit: Int): Query {
+    return limit(null, limit)
+}
+
+/**
+ * Specify the pagination offset parameter of this query.
+ *
+ * This function requires a dialect enabled, different SQLs will be generated with different dialects.
+ *
+ * Note that if [offset] is zero then it will be ignored.
+ */
+public fun Query.offset(offset: Int): Query {
+    return limit(offset, null)
+}
+
 /**
  * Specify the pagination parameters of this query.
  *
  * This function requires a dialect enabled, different SQLs will be generated with different dialects. For example,
  * `limit ?, ?` by MySQL, `limit m offset n` by PostgreSQL.
  *
- * Note that if both [offset] and [limit] are zero, they will be ignored.
+ * Note that if both [offset] and [limit] aren't positive, they will be ignored.
  */
-public fun Query.limit(offset: Int, limit: Int): Query {
-    if (offset == 0 && limit == 0) {
+public fun Query.limit(offset: Int?, limit: Int?): Query {
+    val isOffsetInvalid = offset == null || offset <= 0
+    val isLimitInvalid = limit == null || limit <= 0
+    if (isOffsetInvalid && isLimitInvalid) {
         return this
     }
-
     return this.withExpression(
-        when (expression) {
-            is SelectExpression -> expression.copy(offset = offset, limit = limit)
-            is UnionExpression -> expression.copy(offset = offset, limit = limit)
-        }
+            when (expression) {
```

Comment:
We'd prefer 4-space indent here, but there are 8.

Reply:
Done